app/code/Magento/Backend/Model/Dashboard/Chart.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Backend\Model\Dashboard;
+
+use Magento\Backend\Helper\Dashboard\Data as DataHelper;
+use Magento\Backend\Helper\Dashboard\Order as OrderHelper;
+use Magento\Backend\Model\Dashboard\Chart\Date;
+
+/**
+ * Dashboard chart data retriever
+ */
+class Chart
+{
+    /**
+     * @var Date
+     */
+    private $dateRetriever;
+
+    /**
+     * @var OrderHelper
+     */
+    private $orderHelper;
+
+    /**
+     * @var DataHelper
+     */
+    private $dataHelper;
+
+    /**
+     * Chart constructor.
+     * @param Date $dateRetriever
+     * @param OrderHelper $orderHelper
+     * @param DataHelper $dataHelper
+     */
+    public function __construct(
+        Date $dateRetriever,
+        OrderHelper $orderHelper,
+        DataHelper $dataHelper
+    ) {
+        $this->dateRetriever = $dateRetriever;
+        $this->orderHelper = $orderHelper;
+        $this->dataHelper = $dataHelper;
+    }
+
+    /**
+     * Get chart data by period and chart type parameter, with possibility to pass scope parameters
+     *
+     * @param string $period
+     * @param string $chartParam
+     * @param string|null $store
+     * @param string|null $website
+     * @param string|null $group
+     *
+     * @return array
+     */
+    public function getByPeriod($period, $chartParam, $store = null, $website = null, $group = null): array
```

Comment:
I think it’s good idea to define argument types in all new classes. Will you be able to update your changes?

Reply:
sure, I'll update those. What about changes mentioned by @edenduong, should I move some parts to constants? Keep in mind that i.e. periods are used in few other places I didn't modified yet, and it's hardcoded in all of those, so this won't be easily extendable without rebuilding few other classes.

Reply:
Also functional tests are failing but this is happening since few days to all of my PRs even if I didn't modified anything related - seems those are flaky

Reply:
Can’t we just create all period constants in one class / interface and use in all places ?

Reply:
Sure, that's of course possible. I'll just have to verify few others not yet touched places, I will try to do it today later